### PR TITLE
Added tests for parsing identifiers of various entities.

### PR DESCRIPTION
**Pull request by odisseus**

### CODE REVIEW

File: src/model-test/idtests.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import scala.util.{Success, Try}
+
+object IdTests extends TestApp {
+
+  override def tests(): Unit = {
+
+    test("binary ids") {
+      BinaryId.unapply("foo.bar:baz-quux_2.12:1.2.3-RC4").isEmpty &&
+      BinaryId.unapply("foo.bar:baz-quux_2.12").isEmpty &&
+      BinaryId.unapply("baz-quux_2.12").isDefined
```
Comment (propensive) [inline src/model-test/idtests.scala]: I think that this shouldn't actually be a valid ID. The purpose of the ID is not to encode a version number, and it should be actively discouraged. I'll need to update the code which infers a `BinaryId` from a `BinSpec`, though.

@odisseus Can you update the tests to reflect this? I'll update the method in a separate PR.

**Review by propensive — CHANGES_REQUESTED** [review]

File: src/model-test/idtests.scala (hunk 1 above)
Comment (odisseus) [reply src/model-test/idtests.scala]: Ok

**Comment by propensive** [comment]
Can this be marked "Ready for review" yet?

**Comment by odisseus** [comment]
This pull request can be safely merged now, but I was going to add some more tests before merging.

**Pull request merged**